source/vir/src/context.rs

```diff
@@ -44,6 +45,17 @@ pub struct FunctionCtx {
     pub module_for_chosen_triggers: Option<Path>,
 }
 
+
+/// This stores information about whether a string should be emitted or not. 
+/// To make a string verifiable, one must simply call "reveal" on that string.
+#[derive(Clone, Debug)]
+pub struct VerifiableString {
+    pub inner_str: Arc<String>,
+    pub emitted: bool
+}
+
+pub type GlobalStrings = std::sync::Arc<std::sync::Mutex<HashMap<Path, Arc<VerifiableString>>>>;
```

Comment:
What is this?
StrSlices by default are not lowered into air, at least not in any useful sense.
When there is a "reveal" call, we then write the relevant air to represent that specific string.
This holds all the String values along with their associated variable. So when VAR.reveal() is called, we look up the DefId for VAR and then retrieve the String, from here we can emit the air for this.